src/Persistence.Tests/Extensions/StringExtensionsTest.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.PowerPlatform.PowerApps.Persistence.Extensions;
+
+namespace Persistence.Tests.Extensions;
+
+#pragma warning disable CS8600
+#pragma warning disable CS8604
+
+[TestClass]
+public class StringExtensionsTest
+{
+    [TestMethod]
+    public void FirstCharToUpper_NullInput_ReturnsNull()
+    {
+        string input = null;
+        Assert.IsNull(input.FirstCharToUpper());
```

Comment:
this should throw.
Extension methods should be validating that the `this` parameter is not null before using it.
Calling code should use `input?.FirstCharToUpper()` when input could be null.